packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -505,7 +505,7 @@ class AppDomain extends Domain {
   }
 }
 
-typedef void _DeviceEventHandler(Device device);
+typedef Future<Null> _DeviceEventHandler(Device device);
```

Comment:
It wasn't immediately clear to me what that bit of code did, so I simplified it :) I'll add it back in, w/ perhaps a comment so people understand the pattern it's using.